src/Illuminate/Log/LogManager.php

```diff
@@ -372,9 +373,17 @@ protected function prepareHandlers(array $handlers)
      */
     protected function prepareHandler(HandlerInterface $handler, array $config = [])
     {
-        if (! isset($config['formatter'])) {
+        $isHandlerFormattable = false;
+
+        if (Monolog::API === 1) {
+            $isHandlerFormattable = true;
+        } elseif (Monolog::API === 2 && $handler instanceof FormattableHandlerInterface) {
+            $isHandlerFormattable = true;
+        }
+
+        if ($isHandlerFormattable && ! isset($config['formatter'])) {
             $handler->setFormatter($this->formatter());
-        } elseif ($config['formatter'] !== 'default') {
+        } elseif ($isHandlerFormattable && $config['formatter'] !== 'default') {
```

Comment:
Can't we just check for the `FormattableHandlerInterface` instance everywhere instead of tall the extra if statements and `$isHandlerFormattable` variable?

Reply:
```suggestion
        } elseif ($handler instanceof FormattableHandlerInterface && $config['formatter'] !== 'default') {
```

Reply:
@driesvints 

> elseif ($handler instanceof FormattableHandlerInterface && $config['formatter'] !== 'default')

This would always return `false` on Monolog 1 as the `FormattableHandlerInterface` doesn't exist on Monolog 1.

Reply:
Ah gotcha. nvm